src/flag_gems/ops/index_add.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+import torch
+import triton
+import triton.language as tl
+
+from ..utils import dim_compress, libentry
+
+
+def cfggen():
+    block_m = [1, 2, 4, 8]
+    configs = [
+        triton.Config({"BLOCK_M": m, "BLOCK_N": 1024}, num_warps=4) for m in block_m
+    ]
+    return configs
+
+
+@libentry()
+@triton.autotune(configs=cfggen(), key=["M", "N"])
+@triton.jit
+def index_add_kernel(
+    inp,
+    out,
+    index,
+    src,
+    M,
+    N,
+    alpha,
+    inp_len,
+    BLOCK_M: tl.constexpr,
+    BLOCK_N: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    rows_offsets = pid * BLOCK_M + tl.arange(0, BLOCK_M)[:, None]
+    rows_mask = rows_offsets < M
+
+    for off in range(0, N, BLOCK_N):
+        cols_offsets = off + tl.arange(0, BLOCK_N)
+        index_mask = cols_offsets < N
+        block_mask = rows_mask and index_mask
+
+        cur_indices = tl.load(index + cols_offsets, mask=index_mask, other=0)
+        inp_off = rows_offsets * inp_len + cur_indices[None, :]
+        cur_inp = tl.load(inp + inp_off, mask=block_mask, other=0.0).to(tl.float32)
+        src_off = rows_offsets * N + cols_offsets[None, :]
+        cur_src = tl.load(src + src_off, mask=block_mask, other=0.0).to(tl.float32)

```

Comment:
Possibly lose precision for fp64 src and inputs?

Reply:
What about just keep src and inp as-is without casting?

Reply:
> Possibly lose precision for fp64 src and inputs?

I've encountered precision loss issues in some data types (like bf16 and float32). Ignoring casting might lead to problems. I'll implement the suggested changes below and see if they resolve the issue.